packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -115,198 +118,316 @@ describeNoCompat("GC DataStore Tombstoned When It Is Sweep Ready", (getTestObjec
-
-        // Load a new container and summarizer based on the latest summary
         const {
-            container: summarizingContainer2,
-            summarizer: summarizer2,
-        } = await loadSummarizerAndContainer(summaryVersion);
+            unreferencedId,
+            summarizingContainer,
+            summarizer,
+            summaryVersion,
+        } = await summarizationWithUnreferencedDataStoreAfterTime(sweepTimeoutMs - waitLessThanSweepTimeoutMs);
+
         // Setup close validation
         let closeError: IErrorBase | undefined;
-        summarizingContainer2.on("closed", (error) => {
+        summarizingContainer.on("closed", (error) => {
             closeError = error;
         });
 
         // We load this container from a summary that had not yet tombstoned the datastore so that the datastore loads.
-        const container2 = await provider.loadTestContainer(testContainerConfig, { summaryVersion });
+        const container = await provider.loadTestContainer(testContainerConfig, { summaryVersion });
         // Use the request pattern to get the testDataObject - this is unsafe and no one should do this in their
         // production application. Causes an inactiveObject loaded error
-        const testDataObject2 = await requestFluidObject<ITestDataObject>(container2, testDataObjectId);
+        const dataObject = await requestFluidObject<ITestDataObject>(container, unreferencedId);
 
         // Wait enough time so that the datastore is sweep ready
         await delay(waitLessThanSweepTimeoutMs);
 
-        // Send an op to update the currentTimestampMs to now
-        const mainDataStore2 = await requestFluidObject<ITestDataObject>(summarizingContainer2, "default");
-        mainDataStore2._root.set("send a", "op");
+        await sendOpToUpdateSummaryTimestampToNow(summarizingContainer);
 
         // The datastore should be tombstoned now
-        await summarize(summarizer2);
+        await summarize(summarizer);
 
-        // Send an op - no sweep changed or loaded - the summarizing container does not log sweep ready errors as it
-        // closes before the op is processed and the datastore is realized
-        testDataObject2._root.set("receive", "op");
+        // Send an op to be received - no sweep changed or loaded - the summarizing container does not log sweep ready
+        // errors as it closes before the op is processed and the datastore is realized
+        dataObject._root.set("send an op to be received", "op");
         await provider.ensureSynchronized();
-        assert(summarizingContainer2.closed === true, `Summarizing container should close.`);
+        assert(summarizingContainer.closed === true, `Summarizing container should close.`);
         assert(closeError !== undefined, `Expecting an error!`);
         assert(closeError.errorType === "dataCorruptionError");
         assert(closeError.message === "Context is tombstoned: Call site -  process!");
     });
+
```

Comment:
I simplified the test logic even more to add these handle tests and not drive myself crazy looking through all the different variations. 

Note: if someone makes a request and sets viaHandle = true, we will think it's a call made via handle. We thought about this problem and decided not to do anything about it, as a user can always figure out a way to make the request look like its internal